Framework/DataObjects/test/MDLeanEventTest.h

```diff
@@ -106,6 +107,42 @@ class MDLeanEventTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(a.getSignal(), 2.0);
     TS_ASSERT_EQUALS(a.getError(), 2.0);
   }
+
+  void test_retrieve_coordinates() {
+    constexpr size_t ND(4);
+
+    morton_index::MDSpaceBounds<ND> bounds;
+    // clang-format off
+    bounds <<
+           0.0f, 2.0f,
+        0.0f, 10.0f,
+        3.0f, 7.0f,
+        5.0f, 15.0f;
+    // clang-format on
+
+    morton_index::MDCoordinate<ND> floatCoord;
+    floatCoord << 1.5f, 10.0f, 5.0f, 5.0f;
+
+    MDLeanEvent<ND> event(0.0f, 0.0f, &floatCoord[0]);
+    using EventAccess = MDLeanEvent<ND>::template AccessFor<MDLeanEventTest>;
+#if BOOST_VERSION < MULTIPRECISION_BOOST_VALID_VERSION
+    try {
+      EventAccess::retrieveIndex(event, bounds);
+    } catch (std::runtime_error &exception) {
+      TS_ASSERT_EQUALS(std::string(exception.what()).find("specialisation.") !=
```

Comment:
The idea is to check if the functionality is inaccessible while the boost version is too small: in this case, the exception should be thrown from the interleaving functions, and it is important to avoid using this dummy functions with early boost, this just checks if this safety mechanism is working